src/containers/FileDownload.jsx

```diff
@@ -0,0 +1,87 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { withStyles } from '@material-ui/core/styles';
+import IconButton from '@material-ui/core/IconButton';
+import { searchStyle } from './theme';
+import { connect } from 'react-redux';
+import Download from '@material-ui/icons/CloudDownload';
+
+class FileDownload extends React.Component {
+  constructor(props) {
+    super(props);
+    this.classes = props.classes;
+  }
+
+  exportCsv(){
+    var csvRow = [];
+    var csvData = [['','#','Course Code', 'Course Tittle', 'Link']];
+    var inputCourses = this.props.input;
+    var plannerCourses = this.props.courseData;
+    var tempCourse;
+    var countNum = 1;
+
+    if(plannerCourses){
+      csvData.push(['The courses you have taken:']);
+
+     inputCourses.map(inputCourses =>{
+        tempCourse = inputCourses.split(/(\d+)/);
+        csvData.push([countNum++, tempCourse[0] + ' ' + tempCourse[1] + tempCourse[2]]);
+      });
+
+      csvData.push(['The courses you can take:']);
+
+      countNum = 1;
+
+      plannerCourses.map(plannerCourses =>{
+        csvData.push([countNum++, plannerCourses[0], plannerCourses[1].replace(/,/g, ''), plannerCourses[2]]);  
+      });
+
+      csvData.map(csvData =>{
```

Comment:
`map` is used to transform one array into another without manipulating anything outside the function passed to `map`. `csvData.forEach(csvData => {...})` would be the appropriate function here.